crates/burn-tensor/src/tensor/api/autodiff.rs

```diff
@@ -53,13 +87,13 @@ impl<B: AutodiffBackend> BasicAutodiffOps<B> for Float {
     fn inner<const D: usize>(
         tensor: <Self as TensorKind<B>>::Primitive<D>,
     ) -> <Self::InnerKind as TensorKind<<B as AutodiffBackend>::InnerBackend>>::Primitive<D> {
-        B::inner(tensor)
+        TensorPrimitive::Float(B::inner(tensor.tensor()))
     }
 
     fn from_inner<const D: usize>(
         inner: <Self::InnerKind as TensorKind<<B as AutodiffBackend>::InnerBackend>>::Primitive<D>,
     ) -> <Self as TensorKind<B>>::Primitive<D> {
-        B::from_inner(inner)
+        TensorPrimitive::Float(B::from_inner(inner.tensor()))
```

Comment:
We will need to implement a q_inner and q_from_inner at some point.

Reply:
Oh, this should be fairly straightforward. I'll add it.